### PR TITLE
Add modular navigation baseline implementation

### PR DESCRIPTION
**Pull request by mukulkhanna**
## Motivation and Context

This PR adds the modular navigation baseline implementation from the [home-robot-dev repository](https://github.com/cpaxton/home-robot-dev) to the "Habitat 2.0" rearrange setup. 

## How Has This Been Tested

103997799_171031002_0_radius=0 25

Working with Floorplanner v0.2.0 (exterior-doors closed) scenes with a small (indoor-only) set of 10 episodes shared by @yvsriram. 

Below are some non-ideal and non-default aspects of the implementation:

- Discrete actions (manually updating agent pose)
- Agent looking straight ahead (agent’s pitch angle = 0)
- GT semantics (working with binary semantic mask of goal object)
- 25cm base radius
- Arm is retracted
- 640x480 observations

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/ml/mapping/dense/rotation_utils.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        r2 = np.eye(3)
+    return np.dot(r2, r1)
+
+
+def get_grid(pose, grid_size, precision):
+    """
+    Input:
+        `pose` FloatTensor(bs, 3)
+        `grid_size` 4-tuple (bs, _, grid_h, grid_w)
+        `device` torch.device (cpu or gpu)
```
Comment (Skylion007) [inline habitat-baselines/habitat_baselines/ml/mapping/dense/rotation_utils.py:69]: maybe rename from precision to device? Unclear if you are casting with to over moving between devices

File: habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        camera_elevation_degree : camera elevation to rectify.
+    Output:
+        XYZ : ...x3
+    """
+    R = ru.get_r_matrix(
+        [1.0, 0.0, 0.0], angle=np.deg2rad(camera_elevation_degree)
+    )
+    XYZ = torch.matmul(
+        XYZ.reshape(-1, 3),
+        torch.from_numpy(R).float().transpose(1, 0).to(device),
```
Comment (Skylion007) [inline habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py:76]:
```suggestion
        torch.from_numpy(R).transpose(1, 0).to(device=device, dtype=torch.torch.float32),
```
Do casting and deice transfer at once

File: habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    Input:
+        XYZ                     : ...x3
+        current_pose            : camera position (x, y, theta (radians))
+    Output:
+        XYZ : ...x3
+    """
+    R = ru.get_r_matrix([0.0, 0.0, 1.0], angle=current_pose[2] - np.pi / 2.0)
+    XYZ = torch.matmul(
+        XYZ.reshape(-1, 3),
+        torch.from_numpy(R).float().transpose(1, 0).to(device),
```
Comment (Skylion007) [inline habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py:95]: Ditto, you can do casting and device transfer in same kernel

File: habitat-baselines/habitat_baselines/ml/policy.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        actions = [
+            *envs.call(
+                ["plan_and_step"] * envs.num_envs,
+                [
+                    {"planner_inputs": p_in, "vis_inputs": v_in}
+                    for p_in, v_in in zip(planner_inputs, vis_inputs)
+                ],
+            )
+        ]
+        actions = torch.tensor(actions).to(self.device)
```
Comment (Skylion007) [inline habitat-baselines/habitat_baselines/ml/policy.py:205]:
```suggestion
        actions = torch.tensor(actions, device=self.device)
```

File: habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py (hunk 3 above)
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/ml/mapping/dense/depth_utils.py]: Updated, thanks!

File: habitat-baselines/habitat_baselines/ml/mapping/dense/rotation_utils.py (hunk 1 above)
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/ml/mapping/dense/rotation_utils.py]: Updated, thanks!
